scraper/script.py

```
@@ -0,0 +1,165 @@
+def get_all_recipe_urls():
+    """
+    Fetches the XML sitemap and extracts all recipe URLs.
+    """
+    print(f"Fetching sitemap from: {SITEMAP_URL}...")
+
+    print("Parsing XML...")
+
+    with open("urlset.xml", 'r', encoding='utf-8') as f:
+        response = f.read()
+
+    root = ET.fromstring(response)
```

Ruff (0.14.10)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)